### PR TITLE
First deliverable

### PR DESCRIPTION
**Pull request by AlfredoGJ**

### CODE REVIEW

File: src/components/atoms/Avatar/Avatar.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  ${(props) =>
+    props.shape === 'circle' && {
+      borderRadius: '50%',
+    }}
+
+  ${(props) =>
+    props.shape === 'rounded' && {
+      borderRadius: '1.8rem',
+    }}
+`;
```
Comment (erickwize) [inline src/components/atoms/Avatar/Avatar.jsx]: I recommend extracting all the CSS styling to another file i.e. `Avatar.styled.js`, there export the styles and, in this file use them.

File: src/components/atoms/Grid/Grid.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      return '100%';
+    }};
+  }
+
+  padding-top: ${(props) => props.padding && (props.padding.top || '0rem')}rem;
+  padding-bottom: ${(props) => props.padding && (props.padding.bottom || '0rem')}rem;
+  padding-right: ${(props) => props.padding && (props.padding.right || '0rem')}rem;
+  padding-left: ${(props) => props.padding && (props.padding.left || '0rem')}rem;
+  flex-flow: ${(props) => props.direction};
+`;
```
Comment (erickwize) [inline src/components/atoms/Grid/Grid.jsx]: Same CSS extraction comment here

File: src/components/atoms/Grid/Grid.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    'stretch',
+    'space-between',
+    'space-around'
+  ),
+  xs: PropTypes.oneOf('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'),
+  sm: PropTypes.oneOf('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'),
+  md: PropTypes.oneOf('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'),
+  lg: PropTypes.oneOf('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'),
+  xl: PropTypes.oneOf('1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'),
+};
```
Comment (erickwize) [inline src/components/atoms/Grid/Grid.jsx:45]: if you extract the style you could remove props validation and default values of the component style.

File: src/components/atoms/Icon/Icon.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledIcon = styled.i`
+  color: ${(props) => props.theme.colors[props.color]};
+  font-size: ${(props) => props.size}rem;
+`;
```
Comment (erickwize) [inline src/components/atoms/Icon/Icon.jsx]: Same CSS extraction style comment here

File: src/components/atoms/Avatar/Avatar.jsx (hunk 1 above)
Comment (erickwize) [reply src/components/atoms/Avatar/Avatar.jsx]: If we extract the style from this file we can have a pattern of where styles are written and where logic business is written.

File: src/components/atoms/Input/Input.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const StyledInput = styled.input`
+  width: 100%;
+  outline: none;
+  border: none;
+  background-color: ${(props) => props.theme.colors.backgroundDark};
+  padding: 0.6rem 1rem 0.6rem;
+  border-radius: 8rem;
+  color: ${(props) => props.theme.colors.text};
+  font-family: inherit;
+`;
```
Comment (erickwize) [inline src/components/atoms/Input/Input.jsx]: Same CSS extraction comment here

File: src/components/atoms/Text/Text.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      fontSize: '1rem',
+      color: props.theme.colors.text,
+    }};
+
+  ${(props) =>
+    props.variant === 'body2' && {
+      fontSize: '.8rem',
+      color: props.theme.colors.textLight,
+    }};
+`;
```
Comment (erickwize) [inline src/components/atoms/Text/Text.jsx]: Same CSS extraction comment here

File: src/components/molecules/IconButton/IconButton.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    props.rounded && {
+      width: `${props.size * 2}rem`,
+      height: `${props.size * 2}rem`,
+      borderRadius: '50%',
+    }};
+  ${(props) =>
+    props.background
+      ? `background-color: ${props.theme.colors[props.background]};`
+      : 'background-color:transparent;'}
+`;
```
Comment (erickwize) [inline src/components/molecules/IconButton/IconButton.jsx]: Same CSS extraction comment here

File: src/components/organisms/AppBar/AppBar.jsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  display: block;
+  padding: 0.5rem;
+  position: fixed;
+  width: 100%;
+`;
+
+const StyledSearchIcon = styled(IconButton)`
+  /* margin-left: 2rem;
+  border: 1px solid red; */
+`;
```
Comment (erickwize) [inline src/components/organisms/AppBar/AppBar.jsx]: Same CSS extraction comment here

File: src/components/organisms/DetailedVideoCard/DetailedVideoCard.jsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import Grid from '../../atoms/Grid/Grid';
+import ChannelWidget from '../ChannelWidget/ChannelWidget';
+import Text from '../../atoms/Text/Text';
+
+const CardImage = styled.img`
+  width: 100%;
+  height: 10rem;
+  object-fit: cover;
+  border-radius: 0.8rem;
+`;
```
Comment (erickwize) [inline src/components/organisms/DetailedVideoCard/DetailedVideoCard.jsx]: Same CSS extraction comment here

File: src/components/pages/Home/Home.jsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from 'styled-components';
+import Grid from '../../atoms/Grid/Grid';
+import DetailedVideoCard from '../../organisms/DetailedVideoCard/DetailedVideoCard';
+import mockVideos from '../../../mock/mockVideos';
+
+const StyledHome = styled.div`
+  background-color: ${(props) => props.theme.colors.background};
+`;
```
Comment (erickwize) [inline src/components/pages/Home/Home.jsx]: Same CSS extraction comment here

**Review by erickwize — APPROVED** [review]
## Comments
The code is looking good, I only left some recommendations. I see that you started to implement tests and they look good too.
Please the following deliverables include a file with the coverage of your tests 😊 

## Acceptance Criteria (Mini-challenge 1)
- [x] The header is rendered correctly.
- [x] A list of videos obtained from a mock file is displayed in the **Home View**.
- [x] CSS styles are applied correctly using the styled-components approach.

## Bonus Points (Mini-challenge 1)
- [x] No warnings or errors are logged in the browser console.
- [x] The UI is responsive

## Acceptance Criteria (Mini-challenge 2)
- [x] Meaningful test cases were implemented for components and sub-routines logic.
- [x] All the test cases were successful.

## Bonus Points (Mini-challenge 2)
- [x] Test coverage is above 60%.

File: src/components/atoms/Grid/Grid.jsx (hunk 3 above)
Comment (AlfredoGJ) [reply src/components/atoms/Grid/Grid.jsx:45]: Hello Erick, I added props validation for IntelliSense to remind me the available options, its necesary to remove them?

**Pull request merged**